sweep/walletsweep.go

```diff
@@ -30,6 +30,24 @@ var (
 	ErrFeePreferenceConflict = errors.New("fee preference conflict")
 )
 
+// FeePreference defines an interface that allows the caller to specify how the
+// fee rate should be handled. Depending on the implementation, the fee rate
+// can either be specified directly, or via a conf target which relies on the
+// chain backend(`bitcoind`) to give a fee estimation, or a customized fee
+// function which handles fee calculation based on the specified
+// urgency(deadline).
+type FeePreference interface {
```

Comment:
So is this where the new fee retargeting logic will live?

Reply:
Initially yes. Now it's kept so it's easier to be mocked in tests.